app/table/QueryTable2.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useQuery, type QueryKey, type QueryOptions } from '@tanstack/react-query'
+import { getCoreRowModel, useReactTable, type ColumnDef } from '@tanstack/react-table'
+import React, { useMemo } from 'react'
+
+import { ensurePrefetched, PAGE_SIZE, type ApiError, type ResultsPage } from '@oxide/api'
+
+import { Pagination } from '~/components/Pagination'
+import { usePagination } from '~/hooks/use-pagination'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { TableEmptyBox } from '~/ui/lib/Table'
+
+import { Table } from './Table'
+
+type QueryTableProps<TItem> = {
+  optionsFn: (
+    limit: number,
+    page_token?: string
+  ) => QueryOptions<ResultsPage<TItem>, ApiError> & {
+    queryKey: QueryKey
+  }
+  pageSize?: number
+  rowHeight?: 'small' | 'large'
+  emptyState: React.ReactElement
+  // React Table does the same in the type of `columns` on `useReactTable`
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  columns: ColumnDef<TItem, any>[]
+}
+
+// require ID only so we can use it in getRowId
+export function useQueryTable<TItem extends { id: string }>({
+  optionsFn,
+  pageSize = PAGE_SIZE,
```

Comment:
There's something off that I need to figure out about the fact that `PAGE_SIZE` has a default both here _and_ in `getListQueryOptionsFn`. It means you can manually override the page size in the loader prefetch call and a) it won't apply to the QueryTable, and b) the only way you find out about the mismatch is that the prefetch check in QueryTable fails. It's not the worst (it's more or less what we have now) but it doesn't feel right — it doesn't feel like it's taking advantage of the query options sharing.

I feel like ideally you would not have to specify page size on `useQueryTable` at all; instead, you would set it on the query options object shared by the loader and the query table. It's easy to build the page size into the options object — the only real obstacle is that inside `useQueryTable` we need the `pageSize` for UI reasons (whether to enable the next button), so if we build pageSize into the query options, we need to be able to pull it back out of there. There is a `meta` key on query options that we could put `pageSize` on.